tests/unit/amo/components/TestSearchResult.js

```diff
@@ -131,19 +131,170 @@ describe(__filename, () => {
     });
-    const root = render({ addon });
 
     expect(root).toHaveClassName('SearchResult--theme');
   });
 
+  it('does not render an image if the isAllowedOrigin is false', () => {
+    const differentOriginUrl = 'http://example.com';
+    const isAllowedOriginSpy = sinon.spy();
+    const root = render({
+      isAllowedOrigin: isAllowedOriginSpy,
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+            thumbnail_url: differentOriginUrl,
+          },
+        ],
+      }),
+    });
+
+    sinon.assert.calledWith(isAllowedOriginSpy, differentOriginUrl);
+    expect(root.find('.SearchResult-icon')).toHaveLength(0);
+  });
+
+  it("renders an image's alt attribute as its addon name", () => {
+    const alt = 'pretty image';
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        name: alt,
+      }),
+    });
+
+    expect(root.find('.SearchResult-icon')).toHaveProp('alt', alt);
+  });
+
+  it('renders an empty string for the image alt tag while there is no addon', () => {
+    const root = render({
+      addon: {},
+    });
+
+    expect(root.find('.SearchResult-icon')).not.toHaveProp('alt', '');
+  });
+
+  it('displays the thumbnail image as the default src for static theme', () => {
+    const headerImageThumb =
+      'https://addons.cdn.mozilla.net/mytestthumb/12345.png';
+
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+            thumbnail_url: headerImageThumb,
+          },
+        ],
+      }),
+    });
+    const image = root.find('.SearchResult-icon');
+
+    expect(image.prop('src')).toEqual(headerImageThumb);
+  });
+
+  it('displays srcSet values if preview has multiple options', () => {
+    const headerImageThumb = 'https://addons.cdn.mozilla.net/thumb/12345.png';
+    const headerImageFull = 'https://addons.cdn.mozilla.net/full/54321.png';
+    const thumbWidth = 450;
+    const fullWidth = 900;
+
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        type: ADDON_TYPE_STATIC_THEME,
+        previews: [
+          {
+            ...fakePreview,
+            image_url: 'https://addons.cdn.mozilla.net/thumb/notused.png',
+            thumbnail_url: 'https://addons.cdn.mozilla.net/thumb/notused.png',
+          },
+          {
+            ...fakePreview,
+            image_url: headerImageFull,
+            thumbnail_url: headerImageThumb,
+            image_size: [fullWidth, 200],
+            thumbnail_size: [thumbWidth, 100],
+          },
+        ],
+      }),
+    });
+    const image = root.find('.SearchResult-icon');
+
+    expect(image.prop('srcSet')).toEqual(
+      `${headerImageThumb} ${thumbWidth}w, ${headerImageFull} ${fullWidth}w`,
+    );
+  });
+
+  it.each([0, 1])(
```

Comment:
We actually need to test 3 different things, not just 2. It's a bit more complex, but something like this should work:


```diff
diff --git a/tests/unit/amo/components/TestSearchResult.js b/tests/unit/amo/components/TestSearchResult.js
index 9e00954..6dee78c 100644
--- a/tests/unit/amo/components/TestSearchResult.js
+++ b/tests/unit/amo/components/TestSearchResult.js
@@ -240,9 +240,9 @@ describe(__filename, () => {
     );
   });

-  it.each([0, 1])(
-    `renders image without srcSet if there are no preview image sizes or large image %s`,
-    (index) => {
+  it.each([{ image_url: '' }, { image_size: [] }, { thumbnail_size: [] }])(
+    `renders image without srcSet if there are no preview image sizes or large image %o`,
+    (propOverride) => {
       const headerImageThumb = 'https://addons.cdn.mozilla.net/thumb/12345.png';
       const headerImageFull = 'https://addons.cdn.mozilla.net/full/54321.png';

@@ -258,10 +258,11 @@ describe(__filename, () => {
             },
             {
               ...fakePreview,
-              image_url: index === 0 ? headerImageFull : '',
+              image_url: headerImageFull,
               thumbnail_url: headerImageThumb,
-              image_size: index === 1 ? [400, 200] : [],
-              thumbnail_size: index === 1 ? [200, 100] : [],
+              image_size: [400, 200],
+              thumbnail_size: [200, 100],
+              ...propOverride,
             },
           ],
         }),
```

Reply:
yeh that's so much better 👍